### ISSUE TITLE
Plant Textures

### ISSUE THREAD

**Issue by Ezhh** [Extex101/HW-textures#1]
All textures should be 16*16. Preferred texture license is CC BY-SA 3.0 or CC BY-SA 4.0. If we use any of your work, we can credit you either as Extext101, or whatever name you prefer. 

These are a few things we need for plants. Please let me know if you need more information or colour examples for any of them.

1. Giant mushroom - stem and mushroom top textures. Colours should fit well with the ones here:
https://user-images.githubusercontent.com/7035183/47267664-ea46d780-d53e-11e8-850d-32c6cf91aa64.png (These will be full nodes - similar to giant mushrooms in the ethereal mod.)

2. Desert plants - you can be creative with these. I'd like maybe two types of small plant suitable for deserts. At least one should have a flower. For an idea of colours to match to: https://user-images.githubusercontent.com/7035183/38636541-633797ea-3dc0-11e8-8a9c-5af6542e1ef5.png

3. Spike bush texture. This needs to be dark, probably black. Think barbed wire. It needs to look sharp and it will damage the player. I have tried to make it, but I can't find a good way to do this in 16*16.

4. Jungle grass/bushes/flowers. I'm happy to discuss ideas, because we didn't decide on the ext types needed yet.

I will update this with more another time.

Please let me know if you want to work on any of these or have questions.

**Comment by Extex101** [comment]
![screenshot_2019-02-13-17-46-15-1](https://user-images.githubusercontent.com/33675971/52756866-9feada00-2fb7-11e9-95fa-de04b46fb31b.png)

Spiky bush?

**Comment by Ezhh** [comment]
Much much better than any attempt of mine so far!

**Comment by Extex101** [comment]
![screenshot_2019-02-13-18-17-39-1](https://user-images.githubusercontent.com/33675971/52758106-f22dfa00-2fbb-11e9-9662-d85579829105.png)
Desert plant.. My first attempt.
I can retry if you want, I might get a better result

**Comment by Extex101** [comment]
Also could you give me a copy of that license I can't seem to find it

**Comment by Ezhh** [comment]
For spike bush, these are the colours of things in the same biome:
![shadowystuff](https://user-images.githubusercontent.com/7035183/52781404-601e0400-3044-11e9-9d77-4ae48c3b1c64.png)
It is always night time in this biome (the golden plants give light). So one of the dangers will be that players can't see things that will damage them unless they place light sources.

I like the shape of your spike bush, but could you make the colour match the black grass a little more? I think it needs to be a bit more toward black than brown.

**Comment by Ezhh** [comment]
Hmm... is the desert plant based on a real plant? I am not sure about that one, because I can't tell what the plant is.

**Comment by Extex101** [comment]
No it's not based off of any plant
I can do that if you want

How's this for the spiky
![screenshot_2019-02-14-10-19-40-1](https://user-images.githubusercontent.com/33675971/52808627-2dbfd700-3043-11e9-91aa-7c6f51b6cb91.png)

**Comment by Ezhh** [comment]
That looks good for spiky. I'd be happy to use that. :)

For desert, I'd like it if we can base them on real plants. Hidden Worlds has two types of biome - realistic biomes (forest, glacier, marsh, desert and so on) and some fantasy biomes (shadow, toxic and so on). For the realistic biomes, plants should look like real ones whenever possible. For example, in the grassy plains biome we have bluebells, buttercups and poppies.

**Comment by Extex101** [comment]
Added to the repo

**Comment by Extex101** [comment]
![screenshot_2019-02-14-13-19-00-1](https://user-images.githubusercontent.com/33675971/52818294-37edcf80-305b-11e9-83d8-6d2e21609592.png)
Mushroom stem

**Comment by Extex101** [comment]
![screenshot_2019-02-14-14-24-18-1](https://user-images.githubusercontent.com/33675971/52821675-62905600-3064-11e9-84dc-d786c1d6ca6a.png)
Mushroom

**Comment by Ezhh** [comment]
You are both good at this, and quick. I especially like the look of the stem.

I'll have to test these against the other textures over the weekend, if that's okay.

For licensing, which version of CC are you using?
 I recommend https://creativecommons.org/licenses/by-sa/4.0/

This is the one we are mainly using so far, and it means people must give you credit and must keep the same license terms if reusing your work. Could you take a look and let me know if you would be happy with this?

**Comment by Extex101** [comment]
Isn't that what I have? https://github.com/Extex101/HW-textures/blob/master/LICENCE
Also new desert flower:
Desert Christmas tree
![imag018](https://user-images.githubusercontent.com/33675971/52824019-e437b200-306b-11e9-8989-0bf46dba05f6.JPG)
![screenshot_2019-02-14-15-04-04-1](https://user-images.githubusercontent.com/33675971/52824027-e732a280-306b-11e9-96a6-6484c69ad1f1.png)
It's 32x32 though with a visual_scale of 2 and meshoptions set to 1
![screenshot_2019-02-14-15-06-16-1](https://user-images.githubusercontent.com/33675971/52824031-ea2d9300-306b-11e9-8afc-cea67cd50020.png)
Shorter version in case you don't want it that high

**Comment by Extex101** [comment]
Maybe make it like super rare and I'll have the other two desert plants as well

**Comment by Extex101** [comment]
Also you should make the spiky bushes spawn in groups so it's more realistic and the player can just walk on by without getting hurt

**Comment by Ezhh** [comment]
Hmm, that's an interesting plant. I like that one.

And I see it's CC, but it doesn't say which version, so it's a bit confusing.

**Comment by Extex101** [comment]
OK XD I forgot the title... Fixed:

**Comment by Extex101** [comment]
Also what does your jungle look like?

**Comment by Extex101** [comment]
Oops

**Comment by Extex101** [comment]
Could you give me an example of what your jungle looks like?
So I can get the grass, bush and flowers the right colours

**Comment by Extex101** [comment]
![screenshot_2019-02-15-12-52-15-1](https://user-images.githubusercontent.com/33675971/52883774-ef004e80-3120-11e9-9759-08c5e72a5140.png)
### Desert grass
I tried to duplicate the basic colour of your sand

**Comment by Ezhh** [comment]
Yes - that looks good. I think we can definitely use that grass.

**Comment by Extex101** [comment]
Could you give me an example of what your jungle looks like?
So I can get the grass, bush and flowers the right colours

**Comment by Extex101** [comment]
Honestly just dump everything you have on me and I'll do it one-by-one.
As you said in fast so I can do it.

**Comment by Ezhh** [comment]
You are fast, but I am not - and I need to think things through carefully so I ask for the right things. (I also have a cold/fever/something since yesterday, so it's hard to focus well.)

This should give an idea of  the current jungle:
![jungls](https://user-images.githubusercontent.com/7035183/52914205-987c4700-32bd-11e9-84b6-f97c0fe9aee3.png)
The current trees are giant ones and we'll be adding smaller ones between them, so we need plants that match well with these colours.

Actual jungle surface texture, in case you need to colour match: 
![base_jungle_grass](https://user-images.githubusercontent.com/7035183/52914228-d0838a00-32bd-11e9-9877-a4b2d2a1a0a0.png)

**Comment by Extex101** [comment]
![screenshot_2019-02-17-13-50-17-1](https://user-images.githubusercontent.com/33675971/52919832-37ec0a80-32bb-11e9-9af8-b8c256cef4f2.png)
Jungle grass, bush, and bromeliad flower

**Comment by Extex101** [comment]
![screenshot_2019-02-17-17-48-51-1](https://user-images.githubusercontent.com/33675971/52923818-77781e00-32de-11e9-8f99-48089b2ce696.png)


Purple flowers can be either

![imag005](https://user-images.githubusercontent.com/33675971/52923503-0126ec00-32dd-11e9-83ae-b8fb98c2731e.JPG)
### Scorpion weed /\
Stinks and causes itching
Or
### Glandular Cape Marigold
![imag008](https://user-images.githubusercontent.com/33675971/52923748-1a7c6800-32de-11e9-8c5a-b39d4128a46c.JPG)
But we could just call it "Desert Marigold"

**Comment by Ezhh** [comment]
Is the bromeliad flat? That's the only thing I am not sure about at the moment.

The desert plant set looks brilliant.

**Comment by Extex101** [comment]
Yeah the bromeliad in real life it sorta grows in layers so I guess we could have it grow extra layers using flat node boxes and eventually it could start growing pineapples


And for the people desert flowers which do you think it should be? 
The scorpion weed or desert marigold?

**Comment by Extex101** [comment]
![bromeliad](https://user-images.githubusercontent.com/33675971/52975772-a85a6080-337b-11e9-8ed8-778f34c0d4ac.png)

**Comment by Extex101** [comment]
Pineapple is an original jungle food

**Comment by Extex101** [comment]
![flower under](https://user-images.githubusercontent.com/33675971/52979943-c5e3f600-338c-11e9-9c23-524212645cca.png)
![flower over](https://user-images.githubusercontent.com/33675971/52979945-c67c8c80-338c-11e9-887c-031701e938fe.png)
How's this? 
node_box = {
        type = "fixed",
	fixed = {
		{-0.5, -0.0625, -0.5, 0.5, -0.0625, 0.5},
		{-0.5, -0.5, 0, 0.5, -0.0625, 0},
		{0, -0.5, -0.5, 0, -0.0625, 0.5}
	}
},

**Comment by Ezhh** [comment]
I'd say scorpion weed. It sounds more interesting.

The bromeliad still feels wrong to me. Instead of using a nodebox/having a top texture, I think we'd need a good side texture and then try meshoptions to give a fuller side appearance. (I forget which option it is now, but you can have it so that instead of the plant having two textures making an X shape, it can have four in a box that tapers outward.)

Would you be willing to add a code file with the code for the nodes you've registered as well? The plants mod is LGPLv2.1, so you need to be happy with this license, but it would save time getting these into the game.

**Comment by Extex101** [comment]
Hey should I make several stages of the jungle grass?

**Comment by Extex101** [comment]
Done and uploaded.
Just a few notes:
I called for the textures in a more appropriate way
e.g. "plants_shadow_bush.png", "plants_mushroom_stem_top.png"
So if you use the same code the textures will require some renaming
I didn't add sounds because I doubt you are using the same soundset
Some of the groups are left as 'groups = {cracky=1}' for adjusting to the desired groups 
Spiky/Shadow bush doesn't hurt atm
There is a local variable which equals the current modname 
So the nodes are registered like this:
name..":shadow_bush"
So it can fit into whatever mod you want

If you have any questions just ask

**Comment by Ezhh** [comment]
Thank you. I'll be able to adapt as needed, but it gives me something to work from.

For jungle grass: yes, and we can include a version that extends beyond one nodebox in size as well (check how MTG does Jungle Grass, because I think that's the same).

**Comment by Extex101** [comment]
![screenshot_2019-02-20-12-59-56-1](https://user-images.githubusercontent.com/33675971/53124401-1d5e9f00-3510-11e9-98fb-57c852172586.png)
Tall jungle grass

**Comment by Ezhh** [comment]
How many nodes high is that?

**Comment by Extex101** [comment]
![screenshot_2019-02-20-21-24-30-1](https://user-images.githubusercontent.com/33675971/53145766-57529400-3556-11e9-90c3-ba55b50f7e38.png)

**Comment by Extex101** [comment]
Yeah that might be a bit to much of a size gap....
Added another size
![screenshot_2019-02-21-15-21-24-1](https://user-images.githubusercontent.com/33675971/53208779-e7431d00-35ec-11e9-99f2-f3ebdae98e87.png)

**Comment by Ezhh** [comment]
I think the problem with the largest one is that it looks really stretched. The other two look okay.

My worry is how it will look when there are lots of them together. I may end up making some small adjustments after testing, if that's okay.

**Comment by Extex101** [comment]
Yeah OK I'll go upload them

**Comment by Ezhh** [comment]
Desert plants added. Desert Christmas Tree and Scorpion Weed look great, but I think we need to adjust the dry grass colour just a little to match the sand better (at the bottom).
![desertblah](https://user-images.githubusercontent.com/7035183/53302595-94b96900-3857-11e9-8f35-9b2ff7b987d5.png)

If you are happy to adjust, this is the actual sand texture, if you need it for matching. 
![base_desert_sand](https://user-images.githubusercontent.com/7035183/53302599-b155a100-3857-11e9-9410-7a1a9e57132e.png)

(I've made some adjustments to the code, but I love how you used meshoptions for the Christmas Tree.)

**Comment by Ezhh** [comment]
Added spiky bush (I've decided to name it Dark Thorns instead). This one may also need a slight recolouring. I mostly feel the dark pixels are slightly too absolute black. 

This is the texture  for the surface it will be on: 
![base_shadow_grass](https://user-images.githubusercontent.com/7035183/53303355-5163f800-3861-11e9-86af-bc2262c02d71.png)

**Comment by Extex101** [comment]
OK changed the desert grass, and made the spiky bush's black thorns a bit brighter and added the tinge of red like in your surface texture

### So Is that all the plants you need?

**Comment by Extex101** [comment]
Also how are the mushrooms, and the jungle stuff looking?

**Comment by Ezhh** [comment]
I'm waiting to add the jungle plants for the moment because someone else is working on a tree, and I need to place the plants after the trees.

For mushrooms, I need to create some schematics. 

There will be more plants but not quite yet. Have to think more.
